### PR TITLE
Minor bug fix

### PR DESCRIPTION
**Pull request by AniruddhaKanhere**
Description
-----------
When the `FreeRTOS_sendto` function fails, the status returned by the `prvSendDHCPDiscover` function should be pdFAIL. Instead, a pdPASS was being returned.
This PR fixes that minor bug with an `else` statement.

Additionally, return value description was missing from the Doxygen comments. It was added as well.

Test Steps
-----------

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: FreeRTOS_DHCP.c [hunk 1]
```diff
@@ -1449,8 +1451,10 @@
                  * returned to the stack. */
                 FreeRTOS_ReleaseUDPPayloadBuffer( pucUDPPayloadBuffer );
             }
-
-            xResult = pdTRUE;
+            else
+            {
+                xResult = pdPASS;
+            }
```
Comment (htibosch) [inline FreeRTOS_DHCP.c:1457]: You are right, it makes no sense to wait for a reply to a messages that was never sent.

**Review by htibosch — APPROVED** [review]
@AniruddhaKanhere : thanks for this mini bug-fix. It looks good to me.

File: FreeRTOS_DHCP.c [hunk 2]
```diff
@@ -1449,8 +1451,10 @@
                  * returned to the stack. */
                 FreeRTOS_ReleaseUDPPayloadBuffer( pucUDPPayloadBuffer );
             }
-
-            xResult = pdTRUE;
+            else
```
Comment (gkwicker) [inline FreeRTOS_DHCP.c:1454]: Is there a unit test for this code?  If so, can it be updated to cover this condition?

**Review by gkwicker — COMMENTED** [review]
One question, see comment.

File: FreeRTOS_DHCP.c (hunk 2 above)
Comment (AniruddhaKanhere) [reply FreeRTOS_DHCP.c:1454]: There are no unit-tests as of now. We are working on adding unit-tests to get 100% coverage.

**Review by gkwicker — APPROVED** [review]

**Pull request merged**